tests/test_models.py

```diff
@@ -187,3 +198,143 @@ def test_generate_soil_model(
 
     # Final check that expected logging entries are produced
     log_check(caplog, expected_log_entries)
+
+
+# -------------------------------------
+# abiotic model tests
+# -------------------------------------
+
+
+@pytest.mark.parametrize(
+    "soil_layers,canopy_layers,raises,expected_log_entries",
+    [
+        (
+            2,
+            3,
+            does_not_raise(),
+            (),
+        ),
+        (
+            -2,
+            3,
+            pytest.raises(InitialisationError),
+            (
+                (
+                    CRITICAL,
+                    "There has to be at least one soil layer in the abiotic model!",
+                ),
+            ),
+        ),
+        (
+            2,
+            -3,
+            pytest.raises(InitialisationError),
+            (
+                (
+                    CRITICAL,
+                    "There has to be at least one canopy layer in the abiotic model!",
+                ),
+            ),
+        ),
+        (
+            2.5,
+            3,
+            pytest.raises(InitialisationError),
+            (
+                (
+                    CRITICAL,
+                    "The number of soil layers must be an integer!",
+                ),
+            ),
+        ),
+        (
+            2,
+            3.4,
+            pytest.raises(InitialisationError),
+            (
+                (
+                    CRITICAL,
+                    "The number of canopy layers must be an integer!",
+                ),
+            ),
+        ),
+    ],
+)
+def test_abiotic_model_initialization(
+    caplog, soil_layers, canopy_layers, raises, expected_log_entries
+):
+    """Test `AbioticModel` initialization."""
+
+    with raises:
+        # Initialize model
+        model = AbioticModel(
+            timedelta64(1, "W"), datetime64("2022-11-01"), soil_layers, canopy_layers
+        )
+
+        # In cases where it passes then checks that the object has the right properties
+        assert set(["setup", "spinup", "update", "cleanup"]).issubset(dir(model))
+        assert model.model_name == "abiotic"
+        assert (
+            repr(model)
+            == f"AbioticModel(update_interval = 1 weeks, next_update = 2022-11-08, "
+            f"soil_layers = {int(soil_layers)}, "
+            f"canopy_layers = {int(canopy_layers)})"
+        )
```

Comment:
Do you actually want to check that the `repr()` for your model looks like this or do you just want to check that it has certain properties with certain values? If the latter, I'd write this out more explicitly, even if it means being more verbose.

Reply:
I guess both, I updated the test